src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToArrayTransformer.php

```diff
@@ -72,14 +72,15 @@ public function transform($dateTime)
             ), array_flip($this->fields));
         }
 
-        if (!$dateTime instanceof \DateTime) {
-            throw new TransformationFailedException('Expected a \DateTime.');
+        // DateTimeInterface was introduced in PHP 5.5.0.
+        if (!$dateTime instanceof \DateTime && !$dateTime instanceof \DateTimeInterface) {
+            throw new TransformationFailedException('Expected a \DateTime or \DateTimeInterface.');
         }
 
         $dateTime = clone $dateTime;
         if ($this->inputTimezone !== $this->outputTimezone) {
             try {
-                $dateTime->setTimezone(new \DateTimeZone($this->outputTimezone));
+                $dateTime = $dateTime->setTimezone(new \DateTimeZone($this->outputTimezone));
```

Comment:
@dosten because DateTimeImmutable does not mutate the object but only returns a new one (this is the whole point of having an immutable datetime).
The code works fine for all code because the mutable datetime also returns itself in mutators.